File: Exo/ExoPHP/03_Rectangle/Sphere.php
<?php
require 'Cercle.php';
class Sphere extends Cercle{

/********************Variable*******************/




/********************Accesseur*******************/




/********************Construct*******************/


    public function __construct(array $options = []){
        if (!empty($options)) {
            $this->hydrate($options);
        }
    }


/********************Methode*******************/

    public function perimetre()  {
    }
    public function aire()  {
        return 4*pi()*pow($this->_diametre/2,2);
    }

    public function volume()  {
        return (3/4)*pi()*pow($this->_diametre/2,3);
    }

    public function afficher()  {
        echo $this; 
    }
    public function __toString(){
        $retour = "Diametre : ".$this->_diametre."cm - Périmètre : ".round($this->perimetre(),2)."cm - Aire : ".round($this->aire(),2)."cm² - Volume : ".round($this->volume(),2)."cm³" ;
        return $retour."\n";
    }

}

File: Exo/exoCs/ExoClasse/Exo_Class_04/Sphere.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace Exo_Class_04
{
    internal class Sphere : Cercle
    {
        public Sphere(int diametre) : base(diametre)
        {
        }

        public override string ToString()
        {
            return "Diametre : " + Diametre + " - Périmètre : " + Math.Round(Perimetre(), 2) + " - Volume : " + Math.Round(Volume(), 2);
        }

        public Double Volume()
        {
            return 4/3*double.Pi * Math.Pow(Diametre / 2,3);
        }


        public void AfficherSphere()
        {
            Console.WriteLine(this);
        }
    }
}
